src/connection_string.ts

```diff
@@ -785,26 +779,842 @@ function parseURI(uri: string): { srv: boolean; url: URL; hosts: string[] } {
+  sslCA: {
+    deprecated: true,
+    rename: 'ca',
+    type: 'asIs'
+  },
+  sslCRL: {
+    rename: 'crl',
+    type: 'asIs'
+  },
+  sslCert: {
+    deprecated: true,
+    rename: 'cert',
+    type: 'asIs'
+  },
+  sslKey: {
+    deprecated: true,
+    rename: 'key',
+    type: 'asIs'
+  },
+  sslPass: {
+    deprecated: true,
+    rename: 'passphrase',
+    type: 'string'
+  },
+  sslValidate: {
+    rename: 'rejectUnauthorized',
+    type: 'boolean'
+  },
+  tls: {
+    type: 'boolean'
+  },
+  tlsAllowInvalidCertificates: {
+    type: 'boolean'
+  },
+  tlsAllowInvalidHostnames: {
+    type: 'boolean'
+  },
+  tlsCAFile: {
+    rename: 'ca',
+    type: 'asIs'
+  },
+  tlsCertificateFile: {
+    rename: 'cert',
+    type: 'asIs'
+  },
+  tlsCertificateKeyFile: {
+    rename: 'key',
+    type: 'asIs'
+  },
+  tlsCertificateKeyFilePassword: {
+    rename: 'passphrase',
+    type: 'asIs'
+  },
+  tlsInsecure: {
+    type: 'boolean'
+  },
+  user: {
+    rename: 'credentials',
+    transform({ values: [value], options }) {
+      return new MongoCredentials({ ...options.credentials, username: String(value) });
+    }
+  } as OptionDescriptor,
+  username: {
+    rename: 'credentials',
+    transform({ values: [value], options }) {
+      return new MongoCredentials({ ...options.credentials, username: String(value) });
+    }
+  },
+  validateOptions: {
+    type: 'boolean'
+  },
+  version: {
+    transform({ values: [value], options }) {
+      return { ...options.driverInfo, version: String(value) };
+    }
+  } as OptionDescriptor,
+  w: {
+    rename: 'writeConcern',
+    transform({ values: [value], options }) {
+      return WriteConcern.fromOptions({ ...options.writeConcern, w: value as W });
+    }
+  },
+  waitQueueMultiple: {
+    type: 'uint'
+  },
+  waitQueueTimeoutMS: {
+    type: 'uint'
+  },
+  writeConcern: {
+    rename: 'writeConcern',
+    transform({ values: [value], options }) {
+      if (isRecord(value)) {
+        return WriteConcern.fromOptions({
+          ...options.writeConcern,
+          ...value
+        });
+      }
+      throw new MongoParseError(`WriteConcern must be an object, got ${JSON.stringify(value)}`);
+    }
+  },
+  wtimeout: {
+    rename: 'writeConcern',
+    transform({ values: [value], options }) {
+      const wc = WriteConcern.fromOptions({
+        ...options.writeConcern,
+        wtimeout: getUint('wtimeout', value)
+      });
+      if (wc) return wc;
+      throw new MongoParseError(`Cannot make WriteConcern from wtimeout`);
+    }
+  },
+  wtimeoutMS: {
+    rename: 'writeConcern',
+    transform({ values: [value], options }) {
+      const wc = WriteConcern.fromOptions({
+        ...options.writeConcern,
+        wtimeoutMS: getUint('wtimeoutMS', value)
+      });
+      if (wc) return wc;
+      throw new MongoParseError(`Cannot make WriteConcern from wtimeout`);
+    }
+  },
+  zlibCompressionLevel: {
+    type: 'int'
   }
+} as Record<keyof MongoClientOptions, OptionDescriptor>;
```

Comment:
This is cast after declaration, and explicitly defined in the beginning of the declaration. Is that necessary?

Reply:
Nope, just the cast, fixed!